### DISCUSSION TITLE
have_enqueued_job and queue_with_priority

### DISCUSSION THREAD

**Discussion by mbajur** [rspec/rspec-rails#2757] [General]
Hello!
I have a job that is being enqueued both with `set(priority: 10)` and  defaults to `queue_with_priority 0` and i was wondering if there is any way to test enqueued priority with `have_enqueued_job` matcher? 

According to [this page](https://www.rubydoc.info/gems/rspec-rails/RSpec%2FRails%2FMatchers%3Ahave_enqueued_job) there is not but maybe i'm missing something

Thanks!

**Comment by pirj** [comment]
We do [check the queue](https://github.com/rspec/rspec-rails/blob/a1f991904e8c652faaf241a6549eb82a7d6ff917/lib/rspec/rails/matchers/active_job.rb#L184), but not priority. Do you want to contribute that?

**Reply by mbajur** [reply]
What you think would be a good matcher name? `with_priority`? `having_priority`?

**Reply by mbajur** [reply]
Anyway, see https://github.com/rspec/rspec-rails/pull/2759
